app/src/main/java/com/nervousfish/nervousfish/modules/database/GsonDatabaseAdapter.java

```diff
@@ -104,19 +110,13 @@ public void addContact(final Contact contact) throws IOException {
     public void deleteContact(final String contactName) throws IllegalArgumentException, IOException {
         // Get the list of contacts
         final List<Contact> contacts = this.getAllContacts();
-        final int lengthBefore = contacts.size();
         for (final Contact contact : contacts) {
             if (contactName.equals(contact.getName())) {
                 contacts.remove(contact);
                 break;
             }
         }
 
-        // Throw if the contact to remove is not found
-        if (contacts.size() == lengthBefore) {
```

Comment:
Don't we want to know if the contact was not found and thus could not be deleted?